### PR TITLE
Mafia Achievements

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Winning for the first time as each role gives you an achievement, nice.

There's also a miscellaneous one for getting more votes than the game can hold players, there are a ton of really ODD things that can happen during mafia that I could also make awards for. Don't tempt me!

## Why It's Good For The Game

This was just a no brainer for me honestly (and a nice break from tgui hell), Flex your mighty social deduction skills! Literally black out from blood loss when all your blood goes south upon the unlocking of the mighty nightmare win!

## Changelog
:cl:
add: Mafia Achievements
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
It took me so long to make those achievement icons 😭

**Comment by ShibaInuLord** [comment]
I love armhulen

**Comment by tralezab** [comment]
who now?

File: code/datums/achievements/mafia_achievements.dm [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	icon = "assistant"
+
+/datum/award/achievement/mafia/detective
+	name = "Detective Victory"
+	desc = "If you did this with a Medical Doctor in the game, i'm not really that impressed."
+	database_id = MAFIA_MEDAL_DETECTIVE
+	icon = "detective"
+
+/datum/award/achievement/mafia/psychologist
+	name = "Assistant Victory"
```
Comment (trollbreeder) [inline code/datums/achievements/mafia_achievements.dm:20]: missed this one
```suggestion
	name = "Psychologist Victory"
```

File: code/datums/achievements/mafia_achievements.dm [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+/datum/award/achievement/mafia/traitor
+	name = "Traitor Victory"
+	desc = "Guys, we still have two more changelings to ki-!! TRAITOR VICTORY !!"
+	database_id = MAFIA_MEDAL_TRAITOR
+	icon = "traitor"
+
+/datum/award/achievement/mafia/nightmare
+	name = "Nightmare Victory"
+	desc = "Yeah I got nothing for this one, it's traitor but harder. At least you give thoughtfeeders headaches!"
```
Comment (trollbreeder) [inline code/datums/achievements/mafia_achievements.dm:69]: I'll give you something, based on true events
```suggestion
	desc = "DID YOUR LIGHT FLICKER?!"
```

**Review by trollbreeder — COMMENTED** [review]
minor review of some text related stuff

**Comment by RaveRadbury** [comment]
Next add some achievements for CTF

**Comment by tralezab** [comment]
I need to make some of these work actually :^(

**Comment by tralezab** [comment]
OK. now they should award the person if they are no longer in their body or something, and roles that win conditionally outside of check_victory get their awards

**Comment by Rohesie** [comment]
Tested and working as intended?

**Comment by tralezab** [comment]
Tested as much as I reasonably could! Let the checks pass first but they totally will

**Pull request merged**